backend/pkg/rest/applications_resources_controller.go

```diff
@@ -71,3 +73,162 @@ func (s *Server) GetApplicationResources(params operations.GetApplicationResourc
+	}
+	// List of package IDs that have relations to resource
+	relatedPackageIDs := []string{}
+	for _, pr := range packagesResourcesByResources {
+		relatedPackageIDs = append(relatedPackageIDs, pr.PackageID)
+	}
+
+	return relatedPackageIDs, nil
+}
+
+func (s *Server) getPackagesToDelete(relatedPackageIDs []string) ([]string, error) {
+	// Get packagesResources by package IDs from the updated resource_packages table.
+	packagesResourcesByPackages, err := s.dbHandler.JoinTables().GetResourcePackagesByPackages(relatedPackageIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get remaining packages: %v", err)
+	}
+
+	remainingPackageIDs := []string{}
+	for _, pr := range packagesResourcesByPackages {
+		remainingPackageIDs = append(remainingPackageIDs, pr.PackageID)
+	}
+	// Get packages that had relation only with the deleted resource.
+	// FindUnique returns a list of strings that exist in 'relatedPackageIDs' and not in 'remainingPackageIDs'.
+	packageIDsToDelete := slice.FindUnique(relatedPackageIDs, remainingPackageIDs)
+
+	return packageIDsToDelete, nil
+}
+
+func (s *Server) getRelatedVulnerabilityIDs(packageIDs []string) ([]string, error) {
+	// Get list of vulnerabilities for packages.
+	packageVulnerabilitiesByPackages, err := s.dbHandler.JoinTables().GetPackageVulnerabilitiesByPackages(packageIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get vulnerabilities for packages: %v", err)
+	}
+	relatedVulnerabilityIDs := []string{}
+	for _, pv := range packageVulnerabilitiesByPackages {
+		relatedVulnerabilityIDs = append(relatedVulnerabilityIDs, pv.VulnerabilityID)
+	}
+
+	return relatedVulnerabilityIDs, nil
+}
+
+func (s *Server) getVulnerabilitiesToDelete(relatedVulnerabilityIDs []string) ([]string, error) {
+	// Get packagesVulnerabilities by vulnerability IDs from the updated package_vulnerabilities table.
+	packagesVulnerabilitiesByVulnerabilities, err := s.dbHandler.JoinTables().GetPackageVulnerabilitiesByVulnerabilities(relatedVulnerabilityIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get remaining vulnerabilities: %v", err)
+	}
+
+	remainingVulnerabilityIDs := []string{}
+	for _, pv := range packagesVulnerabilitiesByVulnerabilities {
+		remainingVulnerabilityIDs = append(remainingVulnerabilityIDs, pv.VulnerabilityID)
+	}
+	// Get packages that had relation only with the deleted resource.
+	// FindUnique returns a list of strings that exist in 'relatedVulnerabilityIDs' and not in 'remainingVulnerabilityIDs'.
+	vulnerabilityIDsToDelete := slice.FindUnique(relatedVulnerabilityIDs, remainingVulnerabilityIDs)
+
+	return vulnerabilityIDsToDelete, nil
+}
+
+func (s *Server) deletePackages(packageIDs []string) {
+	for _, pkgID := range packageIDs {
+		pkg, err := s.dbHandler.PackageTable().GetDBPackage(pkgID)
+		if err == nil {
+			if err = s.dbHandler.PackageTable().Delete(pkg); err != nil {
+				log.Errorf("Failed to delete package ID=%s: %v", pkgID, err)
```

Comment:
not sure that we need to get before we delete, if the delete function will just get an ID, i'm pretty sure we can just delete by id.
https://gorm.io/docs/delete.html#Delete-with-primary-key
We can create a DeleteByIDs, and delete in a batch.
```
db.Delete(&users, []int{1,2,3})
```

Reply:
thx